src/main/java/org/springframework/integration/kafka/inbound/KafkaMessageDrivenChannelAdapter.java

```diff
@@ -168,7 +177,7 @@ public AutoAcknowledgingChannelForwardingMessageListener() {
 
 		@Override
 		public void doOnMessage(Object key, Object payload, KafkaMessageMetadata metadata) {
-			KafkaMessageDrivenChannelAdapter.this.sendMessage(toMessage(key, payload, metadata, null));
+			KafkaMessageDrivenChannelAdapter.this.outputChannel.send(toMessage(key, payload, metadata, null));
```

Comment:
With this we don't get so much gain and lose here the `error-channel` flow if that. Plus `trackMessage` within the `MessageHistory` is standard SI feature.